### ISSUE TITLE
Interpolation for function starting with a leading dot

### ISSUE THREAD

**Issue by sgvignali** [r-lib/cli#522]
Hi all,

first of all thank you for developing this package, it's great!

Since the release of  v 3.4.0, the interpolation of functions starting with a leading dot causes an error that didn't occur with v 3.3.0.

Here is how to reproduce the problem:

```R
.my_func <- function(x, y) {
  x + y
}

cli::cli_text("The sum of 2 and 3 is {.my_func(2, 3)}")
```
`Error in doTryCatch(return(expr), name, parentenv, handler) : style`

It looks like the function is interpreted as inline markup.

Thank you in advance,
Sergio

**Comment by gaborcsardi** [comment]
That is correct, from 3.4.0 if a `{}` expression starts with a dot that is interpreted as inline markup. This is to avoid possible ambiguity.

As a workaround, you can put a space before the dot, or put the expression in `()`:
```r
.my_func <- function(x, y) {
  x + y
}

cli::cli_text("The sum of 2 and 3 is { .my_func(2, 3) }")
```

The error message is quite terrible, though, and we need to improve that.

**Comment by sgvignali** [comment]
Perfect, thank you. Yes it took me a while to figure out the problem, a better error message would help.
